cmd/console/deploy/deploy_test.go

```diff
@@ -0,0 +1,237 @@
+package deploy
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"testing"
+
+	"github.com/mia-platform/miactl/factory"
+
+	"github.com/mia-platform/miactl/sdk"
+
+	"github.com/mia-platform/miactl/renderer"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/h2non/gock.v1"
+)
+
+func TestNewDeployCmd(t *testing.T) {
+	const (
+		projectId   = "4h6UBlNiZOk2"
+		revision    = "master"
+		environment = "development"
+		baseURL     = "http://console-base-url/"
+		apiToken    = "YWNjZXNzVG9rZW4="
+	)
+	const expectedPipelineId = 458467
+	expectedPipelineURL := fmt.Sprintf("https://pipeline-url/%d", expectedPipelineId)
+	triggerEndpoint := fmt.Sprintf("api/deploy/projects/%s/trigger/pipeline/", projectId)
+
+	expectedPipeline := sdk.PipelineConfig{
+		ProjectId:   projectId,
+		PipelineId:  expectedPipelineId,
+		Environment: environment,
+	}
+
+	t.Run("successful deploy", func(t *testing.T) {
+		viper.Reset()
+		defer viper.Reset()
+		defer gock.Off()
+
+		gock.New(baseURL).
+			Post(triggerEndpoint).
+			Reply(200).
+			JSON(map[string]interface{}{
+				"id":  expectedPipelineId,
+				"url": expectedPipelineURL,
+			})
+
+		// define from where login command should read config
+		viper.SetConfigFile("/tmp/.miaplatformctl.yaml")
+
+		viper.Set("apibaseurl", baseURL)
+		viper.Set("apitoken", apiToken)
+		viper.Set("project", projectId)
+		viper.WriteConfigAs("/tmp/.miaplatformctl.yaml")
+
+		buf := &bytes.Buffer{}
+		cmd := NewDeployCmd()
+
+		cmd.SetOut(buf)
+		cmd.SetErr(buf)
+		cmd.Flags().Set("environment", environment)
+		cmd.Flags().Set("revision", revision)
+
+		ctx := factory.WithValue(context.Background(), cmd.OutOrStdout())
+		err := cmd.ExecuteContext(ctx)
+		require.NoError(t, err)
+
+		tableRows := renderer.CleanTableRows(buf.String())
+
+		expectedHeaders := "PROJECT ID | DEPLOY ID | VIEW PIPELINE"
```

Comment:
Headers are correct in upper case?

Reply:
I believe that displaying headers in upper case is a configuration of the table rendered. Indeed, also the `get` command writes headers in title case, but tests checks the upper case version.

https://github.com/mia-platform/miactl/blob/master/cmd/get.go#L84
https://github.com/mia-platform/miactl/blob/master/cmd/get_test.go#L228